packages/ember-testing/lib/helpers.js

```diff
@@ -89,9 +89,9 @@ function check(app, selector, context) {
   Ember.assert('To check \'' + selector +
       '\', the input must be a checkbox', type === 'checkbox');
 
-  run(function() {
-    $el.prop('checked', true).change();
-  });
+  if (!$el.prop('checked')) {
+    run('click', app, selector, context);
```

Comment:
This needs to run through `app.testHelpers.click`.  Something like `app.testHelpers.click(selector, context)`.